assemble.py

```diff
@@ -338,7 +340,21 @@ def get_args():
                 preload_progress.tick(complete=True)
                 continue
 
-            image_file_path = factory_client.get_target_system_image(target, args.out_image_dir, preload_progress)
+            try:
+                image_file_path = factory_client.get_target_system_image(target, args.out_image_dir,
+                                                                         preload_progress)
+            except requests.HTTPError as exc:
+                if exc.response.status_code == 404:
+                    # try to download `ota-ext4` image
+                    logger.info("Target's system image in `.wic` format was not found,"
+                                " trying to get an `ext4` image;"
+                                " not found path: " + exc.request.path)
+                    image_file_path = \
+                        factory_client.get_target_system_image(target, args.out_image_dir,
+                                                               preload_progress, format=".ota-ext4")
+                else:
+                    raise
```

Comment:
I had envisioned making this a factory config setting, but I think this actually make sense and is easier for the user - so long as we explain it.

Reply:
Yep, fallback seems easier atm and fulfill the requirement.